File: examples/api/javascript/auth_example.js
/**
 * 青羽写作平台 API - 认证示例
 * 演示如何使用JavaScript/Node.js进行API认证和调用
 */

const axios = require('axios');

class QingyuAPIClient {
  constructor(baseURL = 'http://localhost:9090/api/v1') {
    this.baseURL = baseURL.replace(/\/$/, '');
    this.token = null;
    this.client = axios.create({
      baseURL: this.baseURL,
      headers: {
        'Content-Type': 'application/json'
      }
    });

    // 请求拦截器 - 自动添加Token
    this.client.interceptors.request.use(config => {
      if (this.token) {
        config.headers.Authorization = `Bearer ${this.token}`;
      }
      return config;
    });

    // 响应拦截器 - 统一错误处理
    this.client.interceptors.response.use(
      response => response.data,
      error => {
        console.error('请求失败:', error.message);
        if (error.response) {
          console.error('响应内容:', error.response.data);
        }
        throw error;
      }
    );
  }

  async register(username, email, password, nickname) {
    console.log('\n=== 用户注册 ===');
    try {
      const response = await this.client.post('/auth/register', {
        username,
        email,
        password,
        nickname
      });
      console.log('注册成功:', response);
      return response;
    } catch (error) {
      throw error;
    }
  }

  async login(username, password) {
    console.log('\n=== 用户登录 ===');
    try {
      const response = await this.client.post('/auth/login', {
        username,
        password
      });

      if (response.data && response.data.token) {
        this.token = response.data.token;
        console.log(`登录成功，Token: ${this.token.substring(0, 20)}...`);
      } else if (response.data && response.data.access_token) {
        this.token = response.data.access_token;
        console.log(`登录成功，Token: ${this.token.substring(0, 20)}...`);
      } else {
        console.log('登录成功，但未获取到Token');
      }

      return response;
    } catch (error) {
      throw error;
    }
  }

  async getProfile() {
    console.log('\n=== 获取用户信息 ===');
    try {
      const response = await this.client.get('/user/profile');
      console.log('用户信息:', JSON.stringify(response, null, 2));
      return response;
    } catch (error) {
      throw error;
    }
  }

  async refreshToken() {
    console.log('\n=== 刷新Token ===');
    try {
      const response = await this.client.post('/auth/refresh');
      if (response.data && response.data.token) {
        this.token = response.data.token;
        console.log(`Token刷新成功: ${this.token.substring(0, 20)}...`);
      }
      return response;
    } catch (error) {
      throw error;
    }
  }

  async logout() {
    console.log('\n=== 登出 ===');
    try {
      const response = await this.client.post('/auth/logout');
      this.token = null;
      console.log('登出成功');
      return response;
    } catch (error) {
      throw error;
    }
  }
}

// 主函数 - 演示认证流程
async function main() {
  const client = new QingyuAPIClient();

  try {
    // 1. 用户注册
    console.log('\n【步骤1】用户注册');
    try {
      await client.register(
        'testuser_js',
        'testjs@example.com',
        'SecurePass123!',
        'JavaScript测试用户'
      );
    } catch (error) {
      console.log('注册可能失败（用户已存在）:', error.message);
    }

    // 2. 用户登录
    console.log('\n【步骤2】用户登录');
    await client.login('testuser_js', 'SecurePass123!');

    // 3. 访问受保护的API
    console.log('\n【步骤3】访问受保护的API');
    if (client.token) {
      await client.getProfile();
    } else {
      console.log('未获取到Token，跳过');
    }

    // 4. 刷新Token
    console.log('\n【步骤4】刷新Token');
    if (client.token) {
      await client.refreshToken();
    }

    // 5. 登出
    console.log('\n【步骤5】登出');
    if (client.token) {
      await client.logout();
    }

    console.log('\n=== 认证示例完成 ===');
    return 0;
  } catch (error) {
    console.error('\n错误:', error.message);
    return 1;
  }
}

// 如果直接运行此文件
if (require.main === module) {
  main().then(exitCode => {
    process.exit(exitCode);
  });
}

module.exports = { QingyuAPIClient };

File: examples/api/python/auth_example.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
青羽写作平台 API - 认证示例
演示如何使用Python进行API认证和调用
"""

import requests
import json
from typing import Optional, Dict, Any

class QingyuAPIClient:
    """青羽写作平台API客户端"""

    def __init__(self, base_url: str = "http://localhost:9090/api/v1"):
        self.base_url = base_url.rstrip('/')
        self.token: Optional[str] = None
        self.session = requests.Session()

    def _request(
        self,
        method: str,
        endpoint: str,
        data: Optional[Dict[str, Any]] = None,
        params: Optional[Dict[str, Any]] = None,
        require_auth: bool = False
    ) -> Dict[str, Any]:
        """发送HTTP请求"""
        url = f"{self.base_url}/{endpoint.lstrip('/')}"
        headers = {"Content-Type": "application/json"}

        if require_auth and self.token:
            headers["Authorization"] = f"Bearer {self.token}"

        try:
            response = self.session.request(
                method=method,
                url=url,
                json=data,
                params=params,
                headers=headers
            )
            response.raise_for_status()
            return response.json()
        except requests.exceptions.RequestException as e:
            print(f"请求失败: {e}")
            if hasattr(e.response, 'text'):
                print(f"响应内容: {e.response.text}")
            raise

    def register(self, username: str, email: str, password: str, nickname: str) -> Dict[str, Any]:
        """用户注册"""
        print("\n=== 用户注册 ===")
        data = {
            "username": username,
            "email": email,
            "password": password,
            "nickname": nickname
        }
        result = self._request("POST", "/auth/register", data=data)
        print(f"注册成功: {result}")
        return result

    def login(self, username: str, password: str) -> Dict[str, Any]:
        """用户登录"""
        print("\n=== 用户登录 ===")
        data = {
            "username": username,
            "password": password
        }
        result = self._request("POST", "/auth/login", data=data)

        # 保存token
        if "data" in result:
            self.token = result["data"].get("token") or result["data"].get("access_token")
            if self.token:
                print(f"登录成功，Token: {self.token[:20]}...")
            else:
                print("登录成功，但未获取到Token")

        return result

    def get_profile(self) -> Dict[str, Any]:
        """获取用户信息"""
        print("\n=== 获取用户信息 ===")
        result = self._request("GET", "/user/profile", require_auth=True)
        print(f"用户信息: {json.dumps(result, indent=2, ensure_ascii=False)}")
        return result

    def refresh_token(self) -> Dict[str, Any]:
        """刷新Token"""
        print("\n=== 刷新Token ===")
        result = self._request("POST", "/auth/refresh", require_auth=True)
        if "data" in result and "token" in result["data"]:
            self.token = result["data"]["token"]
            print(f"Token刷新成功: {self.token[:20]}...")
        return result

    def logout(self) -> Dict[str, Any]:
        """登出"""
        print("\n=== 登出 ===")
        result = self._request("POST", "/auth/logout", require_auth=True)
        self.token = None
        print("登出成功")
        return result


def main():
    """主函数 - 演示认证流程"""

    # 创建API客户端
    client = QingyuAPIClient()

    try:
        # 1. 用户注册
        print("\n【步骤1】用户注册")
        try:
            client.register(
                username="testuser_py",
                email="testpy@example.com",
                password="SecurePass123!",
                nickname="Python测试用户"
            )
        except Exception as e:
            print(f"注册可能失败（用户已存在）: {e}")

        # 2. 用户登录
        print("\n【步骤2】用户登录")
        client.login(username="testuser_py", password="SecurePass123!")

        # 3. 访问受保护的API
        print("\n【步骤3】访问受保护的API")
        if client.token:
            client.get_profile()
        else:
            print("未获取到Token，跳过")

        # 4. 刷新Token
        print("\n【步骤4】刷新Token")
        if client.token:
            client.refresh_token()

        # 5. 登出
        print("\n【步骤5】登出")
        if client.token:
            client.logout()

        print("\n=== 认证示例完成 ===")

    except Exception as e:
        print(f"\n错误: {e}")
        return 1

    return 0


if __name__ == "__main__":
    exit(main())
